src/emailjs-tcp-socket.js

```diff
@@ -619,8 +662,16 @@
                 // process the data available on the socket
                 self._onData(readInfo.data);
 
-                // queue the next read
-                self._readLegacySocket();
+                // Queue the next read.
+                // If a STARTTLS handshake might be upcoming, postpone this onto
+                // the task queue so the IMAP client has a chance to call upgradeToSecure;
+                // without this, we might eat the beginning of the handshake.
+                // If we are already secure, just call it (for performance).
+                if (self.ssl) {     // are we secure yet?
+                    self._readLegacySocket();
+                } else {
+                    setZeroTimeout(self._readLegacySocket.bind(self));
```

Comment:
yeah, i'd trust jake archibald with my codebase, too. i'll have a look at the article later today and get back to you, then :)

funny that [this](https://github.com/emailjs/emailjs-tcp-socket/blob/master/src/emailjs-tcp-socket.js#L606-L609) couple of lines change is the workaround they fucked up in the second incarnation of the api. even weirder that they didn't fix it. chrome apps are slowly withering as google (rightfully) pushes to progressive web apps :-\